include/seqan3/io/record.hpp

```diff
@@ -246,16 +246,34 @@ struct record : detail::transfer_template_args_onto_t<field_types, std::tuple>
                   "You must give as many IDs as types to seqan3::record.");
 
     //!\brief (Re-)Initialise all tuple elements with `{}`.
-    void clear()
+    void clear() noexcept(noexcept(clear_impl(*this, std::make_integer_sequence<size_t, field_types::size()>{})))
     {
         clear_impl(*this, std::make_integer_sequence<size_t, field_types::size()>{});
     }
 private:
-    //!\brief Auxiliary function for clear().
+    //!\brief Auxiliary functions for clear().
     template <size_t ...Is>
-    constexpr void clear_impl(base_type & tup, std::integer_sequence<size_t, Is...> const &)
+    static constexpr void clear_impl(base_type & tup, std::integer_sequence<size_t, Is...> const &)
+       noexcept(noexcept((clear_impl2(std::get<Is>(tup)), ...)))
     {
-        ((std::get<Is>(tup) = {}), ...);
+        (clear_impl2(std::get<Is>(tup)), ...);
+    }
+
+    //!\brief Auxiliary functions for clear().
+    template <typename t>
+    //!\cond
+        requires requires (t & v) { v.clear(); }
+    //!\endcond
+    static constexpr void clear_impl2(t & v) noexcept
+    {
+        v.clear();
+    }
+
+    //!\brief Auxiliary functions for clear().
+    template <typename t>
+    static constexpr void clear_impl2(t & v) noexcept(noexcept(t{}))
```

Comment:
```suggestion
    static constexpr void clear_impl2(t & v) noexcept(noexcept(std::declval(t &) = t{}))
```
Even if the default constructor is noexcept, the assignment needn't be.